src/typing/flow_js.ml

```diff
@@ -5297,8 +5297,13 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
     | _, ElemT (use_op, reason_op, (DefT (_, _, ObjT _) as obj), action) ->
       let propref = match l with
       | DefT (reason_x, _, StrT (Literal (_, x))) ->
-          let reason_prop = replace_reason_const (RProperty (Some x)) reason_x in
-          Named (reason_prop, x)
+        let reason_prop = replace_reason_const (RProperty (Some x)) reason_x in
+        Named (reason_prop, x)
+      | DefT (reason_x, _, NumT (Literal (_, (x, _)))) ->
+        let x = Dtoa.ecma_string_of_float x in
+        let reason_prop = replace_reason_const (RProperty (Some x)) reason_x in
+        (* Probably should be different type constructor *)
```

Comment:
Looks like this broke `{[key: number]: any}` reads,
```js
declare var e: { [key: number]: any };

e[0]; // string `0` [1] is incompatible with  number [2].Flow(InferError)
```

Reply:
I think this https://github.com/facebook/flow/pull/2293#issuecomment-261344263 applies to this also, so probably I fixed it wrong